bench_results.py

```diff
@@ -0,0 +1,55 @@
+import re
+import sys
+
+def convert_to_ms(value, unit):
+    if unit == 'µs':
+        return value / 1000
+    elif unit == 'ms':
+        return value
+    elif unit == 's':
+        return value * 1000
+    else:
+        raise ValueError(f"Unknown time unit: {unit}")
+
+def parse_benchmark_file(file_path):
+    machines = {}
+    pattern = r'(\w+)/\w+\s+time:\s+\[\d+(?:\.\d+)?\s(\w+)\s(\d+(?:\.\d+)?)\s(\w+)\s\d+(?:\.\d+)?\s(\w+)\]'
+
+    with open(file_path, 'r') as file:
+        content = file.read()
+        # Clean the file content
+        cleaned_content = re.sub(r'\n(\s+)time:', ' time:', content)
+        for line in cleaned_content.split('\n'):
+            match = re.match(pattern, line)
+            if match:
+                machine, lower_unit, mid_time, mid_unit, upper_unit = match.groups()
+                # Convert mid_time to milliseconds
+                mid_time_ms = convert_to_ms(float(mid_time), mid_unit)
+                machines[machine] = machines.get(machine, 0) + mid_time_ms
+
+    return machines
```

Comment:
`criterion` is capable of emitting JSON IIRC. Maybe that could save you parsing.

Reply:
I didn't know that, it might be useful to improve the script. Thanks!